### PR TITLE
Remove deprecated Vector member functions

### PR DESCRIPTION
**Pull request by masterleinad**
Deprecated in #10398, #11461, and #11487.

### CODE REVIEW

File: include/deal.II/lac/trilinos_vector.h [hunk 1]
```diff
@@ -792,9 +775,10 @@ namespace TrilinosWrappers
        * stored, the second the index of the one past the last one that is
        * stored locally. If this is a sequential vector, then the result will be
        * the pair <code>(0,N)</code>, otherwise it will be a pair
-       * <code>(i,i+n)</code>, where <code>n=local_size()</code> and
-       * <code>i</code> is the first element of the vector stored on this
-       * processor, corresponding to the half open interval $[i,i+n)$
+       * <code>(i,i+n)</code>, where <code>n</code> is the number of elements
+       * stored on this processor and and <code>i</code> is the first element of
+       * the vector stored on this processor, corresponding to the half open
+       * interval $[i,i+n)$
```
Comment (bangerth) [inline include/deal.II/lac/trilinos_vector.h:781]: Thanks for also updating these unrelated things.

**Review by bangerth — APPROVED** [review]

**Comment by bangerth** [comment]
A whole bunch of tests fail:
```
/jenkins/workspace/dealii-mpi_PR-15534/tests/matrix_free/parallel_multigrid_adaptive_05.cc: In function ‘void do_test(const dealii::DoFHandler<dim>&, bool)’:
/jenkins/workspace/dealii-mpi_PR-15534/tests/matrix_free/parallel_multigrid_adaptive_05.cc:542:35: error: ‘class dealii::LinearAlgebra::distributed::Vector<double, dealii::MemorySpace::Host>’ has no member named ‘local_size’; did you mean ‘allocated_size’?
  542 |   for (unsigned int i = 0; i < in.local_size(); ++i)
      |                                   ^~~~~~~~~~
      |                                   allocated_size
/jenkins/workspace/dealii-mpi_PR-15534/tests/matrix_free/parallel_multigrid_adaptive_05.cc: In instantiation of ‘void LaplaceOperator<dim, fe_degree, n_q_points_1d, number>::compute_inverse_diagonal() [with int dim = 2; int fe_degree = 1; int n_q_points_1d = 2; number = double]’:
/jenkins/workspace/dealii-mpi_PR-15534/tests/matrix_free/parallel_multigrid_adaptive_05.cc:123:7:   required from ‘void LaplaceOperator<dim, fe_degree, n_q_points_1d, number>::initialize(const dealii::Mapping<dim>&, const dealii::DoFHandler<dim>&, const dealii::MGConstrainedDoFs&, const std::map<unsigned int, const dealii::Function<dim>*>&, unsigned int, bool) [with int dim = 2; int fe_degree = 1; int n_q_points_1d = 2; number = double]’
/jenkins/workspace/dealii-mpi_PR-15534/tests/matrix_free/parallel_multigrid_adaptive_05.cc:530:3:   required from ‘void do_test(const dealii::DoFHandler<dim>&, bool) [with int dim = 2; int fe_degree = 1; int n_q_points_1d = 2; number = double]’
/jenkins/workspace/dealii-mpi_PR-15534/tests/matrix_free/parallel_multigrid_adaptive_05.cc:653:53:   required from ‘void test() [with int dim = 2; int fe_degree = 1; Number = double]’
/jenkins/workspace/dealii-mpi_PR-15534/tests/matrix_free/parallel_multigrid_adaptive_05.cc:669:22:   required from here
/jenkins/workspace/dealii-mpi_PR-15534/tests/matrix_free/parallel_multigrid_adaptive_05.cc:354:59: error: ‘class dealii::LinearAlgebra::distributed::Vector<double, dealii::MemorySpace::Host>’ has no member named ‘local_size’; did you mean ‘dealii::LinearAlgebra::distributed::Vector<double, dealii::MemorySpace::Host>::size_type dealii::LinearAlgebra::distributed::Vector<double, dealii::MemorySpace::Host>::allocated_size’? (not accessible from this context)
  354 |     for (unsigned int i = 0; i < inverse_diagonal_entries.local_size(); ++i)
      |                                  ~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~

...
```

**Comment by bangerth** [comment]
The following tests still fail:
```
[projectroot.tests.mpi.p4est_2d_constraintmatrix_03debug.mpirun=4.mpi/p4est_2d_constraintmatrix_03.mpirun=4.debug](https://ci.tjhei.info/job/dealii-mpi/job/PR-15534/3/testReport/junit/projectroot.tests.mpi.p4est_2d_constraintmatrix_03debug/mpirun=4/mpi_p4est_2d_constraintmatrix_03_mpirun_4_debug/)
[projectroot.tests.mpi.p4est_3d_constraintmatrix_02debug.mpirun=4.mpi/p4est_3d_constraintmatrix_02.mpirun=4.debug](https://ci.tjhei.info/job/dealii-mpi/job/PR-15534/3/testReport/junit/projectroot.tests.mpi.p4est_3d_constraintmatrix_02debug/mpirun=4/mpi_p4est_3d_constraintmatrix_02_mpirun_4_debug/)
[projectroot.tests.mpi.p4est_3d_constraintmatrix_02debug.mpirun=10.mpi/p4est_3d_constraintmatrix_02.mpirun=10.debug](https://ci.tjhei.info/job/dealii-mpi/job/PR-15534/3/testReport/junit/projectroot.tests.mpi.p4est_3d_constraintmatrix_02debug/mpirun=10/mpi_p4est_3d_constraintmatrix_02_mpirun_10_debug/)
[projectroot.tests.mpi.p4est_2d_constraintmatrix_03debug.mpirun=10.mpi/p4est_2d_constraintmatrix_03.mpirun=10.debug](https://ci.tjhei.info/job/dealii-mpi/job/PR-15534/3/testReport/junit/projectroot.tests.mpi.p4est_2d_constraintmatrix_03debug/mpirun=10/mpi_p4est_2d_constraintmatrix_03_mpirun_10_debug/)
```
Please squash the commits once you have these tests fixed up!

**Comment by bangerth** [comment]
Still three failing tests:
```
[projectroot.tests.trilinos.readwritevector_03debug.mpirun=2.trilinos/readwritevector_03.mpirun=2.debug](https://ci.tjhei.info/job/dealii-mpi/job/PR-15534/4/testReport/junit/projectroot.tests.trilinos.readwritevector_03debug/mpirun=2/trilinos_readwritevector_03_mpirun_2_debug/)
[projectroot.tests.trilinos.print_01debug.mpirun=2.trilinos/print_01.mpirun=2.debug](https://ci.tjhei.info/job/dealii-mpi/job/PR-15534/4/testReport/junit/projectroot.tests.trilinos.print_01debug/mpirun=2/trilinos_print_01_mpirun_2_debug/)
[projectroot.tests.trilinos.readwritevector_02debug.mpirun=2.trilinos/readwritevector_02.mpirun=2.debug](https://ci.tjhei.info/job/dealii-mpi/job/PR-15534/4/testReport/junit/projectroot.tests.trilinos.readwritevector_02debug/mpirun=2/trilinos_readwritevector_02_mpirun_2_debug/)
```

**Review by kronbichler — APPROVED** [review]

**Pull request merged**